app/views/helpers/injected/obligations/NextUpdatesHelper.scala.html

```diff
@@ -69,27 +70,45 @@
 @getSectionHeadingMessage(obligations.head.obligation)
 }
 
-@bodyContent(collectedObligations: Seq[NextUpdateModelWithIncomeType]) = @{
-    govukTable(Table(
-        rows = {
-            for(obligation <- collectedObligations) yield deadlineRow(obligation)
-        },
-        head = Some(Seq(
-            HeadCell(
-                content = Text(messages("nextUpdates.table.header.type"))
-            ),
-            HeadCell(
-                content = Text(messages("nextUpdates.table.header.source"))
+@makeTable(collectedObligations: Seq[NextUpdateModelWithIncomeType]) = @{
+    collectedObligations match {
+        case validObligations: Seq[NextUpdateModelWithIncomeType] if validObligations.nonEmpty =>
+            govukTable(Table(
+                rows = {
+                    for(obligation <- validObligations) yield deadlineRow(obligation)
+                },
+                head = Some(Seq(
+                    HeadCell(
+                        content = Text(messages("nextUpdates.table.header.type"))
+                    ),
+                    HeadCell(
+                        content = Text(messages("nextUpdates.table.header.source"))
+                    )
+                )),
+                caption = {
+                    if(validObligations.head.obligation.obligationType == "Quarterly")
```

Comment:
consider next change to make this code more safe with respect to data in the collection:

```
caption = {
                    validObligations.headOption.map(x => x.obligation.obligationType) match {
                        case Some("Quarterly") =>
                            validObligations.headOption.map(x => getTableHeadingMessage(x.obligation))
                        case _ => None
                    }
                },
```